src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const resultArr = [];
+
+  for (const { type, extraData, keysToRemove } of actions) {
+    switch (type) {
+      case 'addProperties': {
+        Object.assign(newState, extraData);
+        break;
+      }
+
+      case 'removeProperties': {
+        for (let j = 0; j < keysToRemove.length; j++) {
```

Comment:
As mentioned Oleksandr:
- It is better to replace it with the for...of loop for better readability